File: src/vyrion/_harness/samples/agno/payment_agent.py
"""Sample Agno agent with a requires_confirmation tool."""
from agno.agent import Agent
from agno.tools import tool


@tool(requires_confirmation=True)
def transfer_funds(recipient: str, amount: int, currency: str) -> str:
    """Protected action requiring user confirmation."""
    return "executed"


agent = Agent(name="payments", tools=[transfer_funds])

File: src/vyrion/_harness/samples/vercel_ai/payment-agent.ts
// Sample Vercel AI SDK tool needing human approval before execution.
import { tool } from 'ai';
import { z } from 'zod';

export const transferFunds = tool({
  description: 'Protected action: execute a wire transfer',
  parameters: z.object({ recipient: z.string(), amount: z.number(), currency: z.string() }),
  // needsApproval gates execution; the approval decision is persisted with the
  // tool call and read back on resume.
  needsApproval: true,
  execute: async ({ recipient, amount, currency }) => {
    return { executed: true };
  },
});
